src/app/App.ts

```diff
@@ -0,0 +1,209 @@
+import { a, mount, patch, reactive, watchEffect } from '../shared/lib/act'
+import type { VNode } from '../shared/lib/act'
+
+import './App.styles.css'
+
+import { icons } from './config'
+
+import winSound from '../shared/assets/sounds/win-sound.wav'
+
+import { GameCard, ToggleDiffuculty, WinBanner } from '../components'
+import { shuffleArray, toTimerFormat } from './helpers'
+
+import type { CardPairState } from '../shared/types'
+import type { Card } from '../shared/types/types'
+import { storage } from '../shared/lib/localStorage'
+
+type AppState = {
+  theme: 'light' | 'dark'
+  timerId: number
+  timerValue: number
+  isHard: boolean
+  isPlay: boolean
+  isWinMessage: boolean
+  cardsList: Card[]
+  pair: CardPairState[]
+  matchedCount: number
```

Comment:
название полей не достаточно очевидные и понятные
```suggestion
  isModeHard: boolean
  isPlayInProgress: boolean
  isWinMessageShown: boolean
  cardsList: Card[]
  pair: CardPairState[]
  matchedCardsCount: number
```